flink-core/src/main/java/org/apache/flink/core/fs/ConsistencyLevel.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.fs;
+
+/**
+ * An enumeration describing the level of consistency offered by a {@link FileSystem}.
+ * 
+ * <p>The consistency levels described here make statements about the visibility
+ * of file existence and file contents in the presence of <i>new file creation</i>,
+ * <i>file deletion</i>, <i>file renaming</i>, and <i>directory listing</i>.
+ * 
+ * <p>An operation is defined as consistent if the following holds: After the function
+ * call triggering the operation returns, its result is immediately reflected in
+ * in the view presented to any other party calling a file system function.
+ * 
+ * <p>Please note that these levels do not make any statements about the effects or visibility of
+ * file content modification or file appends. In fact, content modification or appending are
+ * not supported in various file systems.
+ * 
+ * <p>Some of these consistency levels indicate that the storage system does not actually
+ * qualify to be called a FileSystem, but rather a blob-/object store.
+ */
+public enum ConsistencyLevel {
```

Comment:
I understand what you are saying. Initially I had the consistency characteristics in the `FileSystemKind`. I played around with that for a bit and felt that factoring them out is actually the right way to go.

We could change the enum to a class so that we don't have to create a new enum constant for each combination of consistency attributes (I think there will be no too many more, there is a pretty clear hierarchy from file creation --> file rename/delete  --> directory view consistency).

We could also extend the `ConsistencyLevel` to have methods like `supportsConsistentRename()`, which would make it easier to evaluate if a certain property is provided in the using code.